### small image 분류 성능 지표

자전거

In [ ]:
from sklearn.metrics import classification_report

df_bicycle = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_bicycle.csv', encoding='cp1252')
df_bicycle = df_bicycle.rename(columns={'0': 'Predicted', '0.1': 'Actual'})

y_pred = df_bicycle['Predicted']
y_true = df_bicycle['Actual']

report_bicycle = classification_report(y_true, y_pred)
print(report_bicycle)

              precision    recall  f1-score   support

           0       0.96      0.93      0.95       739
           1       0.73      0.84      0.78       160

    accuracy                           0.92       899
   macro avg       0.85      0.89      0.86       899
weighted avg       0.92      0.92      0.92       899



In [ ]:
df_bicycle = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_bicycle.csv', encoding='cp1252')
df_bridge = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_bridge.csv', encoding='cp1252')
df_bus = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_bus.csv', encoding='cp1252')
df_car = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_car.csv', encoding='cp1252')
df_chimney = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_chimney.csv', encoding='cp1252')
df_crosswalk = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_crosswalk.csv', encoding='cp1252')
df_hydrant = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_hydrant.csv', encoding='cp1252')
df_motorcycle = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_motorcycle.csv', encoding='cp1252')
df_palm = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_palm.csv', encoding='cp1252')
df_stair = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_stair.csv', encoding='cp1252')
df_traffic_light = pd.read_csv('/content/drive/MyDrive/Kaggle/Output/df_traffic_light.csv', encoding='cp1252')

In [ ]:
from sklearn.metrics import classification_report
def change_df(df):
  df = df.rename(columns={'0': 'Predicted', '0.1': 'Actual'})
  y_pred = df['Predicted']
  y_true = df['Actual']
  return classification_report(y_true, y_pred)

In [ ]:
report_bicycle = change_df(df_bicycle)
report_bridge = change_df(df_bridge)
report_bus = change_df(df_bus)
report_car = change_df(df_car)
report_chimney = change_df(df_chimney)
report_crosswalk = change_df(df_crosswalk)
report_hydrant = change_df(df_hydrant)
report_motorcycle = change_df(df_motorcycle)
report_palm = change_df(df_palm)
report_stair = change_df(df_stair)
report_traffic_light = change_df(df_traffic_light)

각 클래스당 성능

In [ ]:
# Calculate overall performance
reports = [report_bicycle, report_bridge, report_bus, report_car, report_chimney, report_crosswalk, report_hydrant, report_motorcycle, report_palm, report_stair, report_traffic_light]
class_names = ['bicycle', 'bridge', 'bus', 'car', 'chimney', 'crosswalk', 'hydrant', 'motorcycle', 'palm','stair', 'traffic_light']
overall_report = {'precision': 0, 'recall': 0, 'f1-score': 0, 'support': 0}

for report, class_name in zip(reports, class_names):
  print(class_name)
  print(report)
  print('*'*60)

bicycle
              precision    recall  f1-score   support

           0       0.96      0.93      0.95       739
           1       0.73      0.84      0.78       160

    accuracy                           0.92       899
   macro avg       0.85      0.89      0.86       899
weighted avg       0.92      0.92      0.92       899

************************************************************
bridge
              precision    recall  f1-score   support

           0       0.78      0.94      0.86       567
           1       0.85      0.55      0.67       332

    accuracy                           0.80       899
   macro avg       0.82      0.75      0.76       899
weighted avg       0.81      0.80      0.79       899

************************************************************
bus
              precision    recall  f1-score   support

           0       0.78      0.98      0.87       618
           1       0.92      0.39      0.54       281

    accuracy                           0.

전체 클래스 성능 - 각 클래스에 대한 classification_report
총합 평균


| Class          | Accuracy | F1-score about about class | F1-score except class | count |
|----------------|----------|------------------|------------------|--|
| Bicycle        |   0.92   |      0.95        |      0.78        |899|
| Bridge         |   0.80   |      0.86        |      0.67        |899|
| Bus            |   0.80   |      0.87        |      0.54        |899|
| Car            |   0.90   |      0.93        |      0.81        |899|
| Chimney        |   0.92   |      0.95        |      0.72        |647|
| Crosswalk      |   0.81   |      0.87        |      0.67        |900|
| Hydrant        |   0.92   |      0.94        |      0.88        |899|
| Motorcycle     |   0.87   |      0.92        |      0.58        |728|
| Palm           |   0.88   |      0.91        |      0.81        |900|
| Stair          |   0.78   |      0.87        |      0.29        |899|
| Traffic Light  |   0.92   |      0.95        |      0.81        |899|
| Total          |   0.87   |      0.91        |      0.68        |9468|
